Update Baxter for Kinetic

davetcoleman opened the pull request:
Combines https://github.com/ros-planning/moveit_robots/pull/53 and https://github.com/ros-planning/moveit_robots/pull/52 because there was a merge conflict with those, and fixes some more deprecation problems with kinetic

Note Rethink only supports Indigo and will not officially review or support kinetic or this PR, so I think it is fine if another moveit maintainer merges it if needed @IanTheEngineer 

*This PR is sponsored by [Vicarious](http://www.vicarious.com/)*

davetcoleman committed:
Baxter: Change virtual_joint to base
Baxter: Remove duplicate baxter.srdf
Baxter: Sync with latest setup assistant templates

davetcoleman committed:
Fix Baxter setup assistant roslaunch to work with xacro files

davetcoleman committed:
Fix deprecation MoveGroupExecutionService

davetcoleman committed:
Disable mongodb warehouse_ros functionality because not available via binary

v4hn reviewed (approved):
Looks good to me.
CI for this repository seems to be broken though.

Could you have a look before merging please?

davetcoleman committed:
Updates for Baxter MoveIt! config
 - Disable always in collision link between head and display
 - Make world_joint passive to stop MoveIt! warning
 - Move octomap settings to always be present to suppress warnings
 - Update Rviz config

IanTheEngineer commented on baxter/baxter_moveit_config/config/baxter_base.srdf.xacro:
```diff
@@ -52,7 +52,9 @@
     <!--END EFFECTOR: Purpose: Represent information about an end effector.-->
      <!--VIRTUAL JOINT: Purpose: this element defines a virtual joint between a robot link and an external frame of reference (considered fixed with respect to the robot)-->
-    <virtual_joint name="world_joint" type="floating" parent_frame="world" child_link="torso" />
+    <virtual_joint name="world_joint" type="floating" parent_frame="world" child_link="base" />
```
Does this still allow for two arms to be used in the MoveIt GUI? I vaguely remember I had to change it to `torso` in order to enable both arms

davetcoleman replied:
The virtual joint should connect to the root of the URDF tree. I tested this yesterday and was able to plan with both arms repeatedly.

IanTheEngineer replied:
I'll give this test a whirl as well. Might either have been an issue with end effectors or possibly this was a bug that was fixed in recent updates to MoveIt... Regardless, if it works, I'm fine moving to the `base`  frame.

IanTheEngineer commented on baxter/baxter_moveit_config/launch/demo_baxter.launch:
```diff
@@ -34,5 +34,10 @@
-  <include file="$(find baxter_moveit_config)/launch/default_warehouse_db.launch" />
+  <!-- As of ROS Kinetic / Ubuntu 16.04 the mongo driver does not have a binary and must be installed from source.
+       Therefore, this functionality is disabled by default. For more info:
+       https://github.com/ros-planning/moveit/issues/123
```
+1 very helpful comment

IanTheEngineer reviewed (commented):
Provided that `base` still functions as the `child_link` inside the `virtual_joint`, these changes look good.

IanTheEngineer commented:
The `ci` seems to output a funny message about a non-existent package:
[`[build] Given package '--no-deps atlas_moveit_config
atlas_v3_moveit_config
baxter_ikfast_left_arm_plugin
baxter_ikfast_right_arm_plugin
baxter_moveit_config
iri_wam_moveit_config
moveit_robots
r2_moveit_generated' is not in the workspace`](https://travis-ci.org/ros-planning/moveit_robots/jobs/215061861#L523-L530)

davetcoleman committed:
Change baxter urdf to kinetic-devel branch

IanTheEngineer commented:
I'm fine with merging these changes. @v4hn & @130s, should we be bothered at all with the `ci-build` failing? I don't see how Dave's changes could have broken `ci`.

IanTheEngineer commented:
I tested your branch out on my laptop and get one error, one warning. Neither of them seem to affect performance:
```
ERROR: cannot launch node of type [warehouse_ros/mongo_wrapper_ros.py]: can't locate node [mongo_wrapper_ros.py] in package [warehouse_ros]
[ INFO] [1490623607.735298317]: Loading robot model 'baxter'...
[ WARN] [1490623607.735562021]: Skipping virtual joint 'world_joint' because its child frame 'base' does not match the URDF frame 'world'
```

130s commented:
> The `ci` seems to output a funny message about a non-existent package:
> [`[build] Given package '--no-deps atlas_moveit_config
> atlas_v3_moveit_config
> baxter_ikfast_left_arm_plugin
> baxter_ikfast_right_arm_plugin
> baxter_moveit_config
> iri_wam_moveit_config
> moveit_robots
> r2_moveit_generated' is not in the workspace`](https://travis-ci.org/ros-planning/moveit_robots/jobs/215061861#L523-L530)

This is weird. I tried to reproduce locally but this didn't occur and the corresponding [catkin build](https://github.com/ros-planning/moveit_ci/blob/d11d7d0b45cebf77430adc29b05e36e9db3dee22/travis.sh#L161) command passed without any error. I re-started Travis to see if the same happens.
(If the job ran on Sunday night west US time, I saw Travis/GitHub behaved strange).

davetcoleman commented:
I worked on fixing the CI last week but its having a weird error on running the tests that do not show any errors. I haven't been able to figure out why yet, even after running the CI locally...

davetcoleman committed:
Change branch of moveit_ci for debugging

davetcoleman commented:
I have been working on fixing the CI for several days now and will address the issue (seems to me the line breaks are the problem) but I need to merge this issue so I can make an announcement about Baxter/Gazebo working.
